apps/src/code-studio/viewAsRedux.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Reducer for tracking whether we're currently viewing the page as a student
+ * or as a teacher
+ */
+
+import { makeEnum } from '@cdo/apps/utils';
+import { queryParams, updateQueryParam } from '@cdo/apps/code-studio/utils';
+
+export const ViewType = makeEnum('Student', 'Teacher');
+
+// Action types
+export const SET_VIEW_TYPE = 'viewAs/SET_VIEW_TYPE';
+
+/**
+ * Stage lock reducer
+ */
+export default function reducer(state = ViewType.Student, action) {
+  if (action.type === SET_VIEW_TYPE) {
+    const viewType = action.viewType;
+    if (!ViewType[viewType]) {
+      throw new Error('unknown ViewType: ' + viewType);
+    }
+
+    return action.viewType;
+  }
+
+  return state;
+}
+
+// Action creators
+
+export const setViewType = viewType => {
+  return dispatch => {
+    // If changing to viewAs student while we are a particular student, remove
+    // the user_id and do a reload so that we're instead viewing as a generic
+    // student
+    if (viewType === ViewType.Student && queryParams('user_id')) {
+      updateQueryParam('user_id', undefined);
+      window.location.reload();
+      return;
```

Comment:
Should also test this updateQueryParam path, even if you stub `queryParams` and `updateQueryParam` and `window.location.reload` - you'd still be verifying that it's triggered under the correct conditions, that those stubbed methods get called, and that we don't actually dispatch `SET_VIEW_TYPE` in this case.

It may be helpful to use [the `reload` utility](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/utils.js#L661-L663) for stubbing purposes.